### PR TITLE
Scan new tables from CSVs all the time

### PR DESCRIPTION
**Pull request by tsmacdonald**
<strike>[waiting for the OK from product]</strike> _Approved by Conor_

### CODE REVIEW

**Review by calherries — APPROVED** [review]
Looks good. I just realised there are no tests checking the table actually syncs and gets scanned though. 

Maybe you could add some tests checking basics like:
- does the table have initial_sync_status = "complete"
- do the fields get analyzed

**Comment by tsmacdonald** [comment]
@calherries `with-redefs` is always a little cheeky, but since there's an existing suite of tests to make sure `sync-table!` works (and I just updated the tests to reflect the new `initial_sync_status` last week) I opted for trying to keep the CSV tests more unit-y and mocking out the actual details of table syncing. We do (and have been) testing to make sure fields are created properly though.

I recommend looking at the diff with whitespace changes ignored: https://github.com/metabase/metabase/pull/31862/files?w=1

**Comment by tsmacdonald** [comment]
it's definitely white-box testing to muddle with `future` like that (as I said, cheeky) but I definitely don't want unit tests to have to actually worry about multiple threads.

**Comment by calherries** [comment]
Is it a bit too cheeky? You've tested that `sync/scan-table!` is called inside a future, but I think it's worth doing more. `sync/sync-table!` can change, and this test makes the implementation of `upload-from-csv!` harder to change (we might want to move away from `sync/scan-table!` to using some other functions, in which case we'd need to update the test even though the outcome is the same). I would feel better if we tested some basic properties of the fields that prove the table has been analyzed etc, e.g. checking that a field with name last_name has semantic type `:type/Name`.

**Comment by tsmacdonald** [comment]
fair enough; updated

File: test/metabase/api/card_test.clj [hunk 1]
```diff
@@ -2833,17 +2831,23 @@
...
+                           ["name" :type/Name]}
                          (->> (t2/select Field :table_id (:id new-table))
-                              (map (comp u/lower-case-en :name))
-                              set)))
-                  (is (= {:table-id   (:id new-table)
-                          :in-future? true}
-                         @sync-sentinel)
+                              (map (fn [field] [(u/lower-case-en (:name field))
+                                                (:semantic_type field)]))
+                              set))
```
Comment (calherries) [inline test/metabase/api/card_test.clj:2848]: test description?

**Review by calherries — APPROVED** [review]

File: test/metabase/api/card_test.clj (hunk 1 above)
Comment (calherries) [reply test/metabase/api/card_test.clj:2848]: Oh wow somehow I didn't see that

**Pull request merged**